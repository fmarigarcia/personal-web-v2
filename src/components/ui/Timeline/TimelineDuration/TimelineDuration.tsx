import React from 'react';
import { useTranslation } from 'react-i18next';
import { clsx } from 'clsx';

interface TimelineDurationProps {
    duration: string;
    isSelected: boolean;
}

export const TimelineDuration: React.FC<TimelineDurationProps> = ({
    duration,
    isSelected,
}) => {
    const { t } = useTranslation();

    return (
        <div className="text-center">
            <p
                className={clsx(
                    'text-xs font-medium transition-colors duration-300',
                    {
                        'text-stone-950 dark:text-green-400': isSelected,
                        'text-stone-600 dark:text-zinc-400 group-hover:text-stone-950 dark:group-hover:text-green-400':
                            !isSelected,
                    }
                )}
            >
                {t(duration)}
            </p>
        </div>
    );
};
