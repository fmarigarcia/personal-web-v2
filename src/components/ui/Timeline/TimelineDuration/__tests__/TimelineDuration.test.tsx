import { render, screen } from '@testing-library/react';
import { TimelineDuration } from '../TimelineDuration';

// Mock react-i18next
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: jest.fn((key: string) => key),
    }),
}));

describe('TimelineDuration', () => {
    const defaultProps = {
        duration: 'test.duration.key',
        isSelected: false,
    };

    it('should render duration text', () => {
        render(<TimelineDuration {...defaultProps} />);

        expect(screen.getByText('test.duration.key')).toBeInTheDocument();
    });

    it('should apply selected styles when isSelected is true', () => {
        render(<TimelineDuration {...defaultProps} isSelected={true} />);

        const durationElement = screen.getByText('test.duration.key');
        expect(durationElement).toHaveClass('text-stone-950');
    });

    it('should apply unselected styles when isSelected is false', () => {
        render(<TimelineDuration {...defaultProps} isSelected={false} />);

        const durationElement = screen.getByText('test.duration.key');
        expect(durationElement).toHaveClass(
            'text-stone-600',
            'group-hover:text-stone-950'
        );
    });

    it('should have correct base classes', () => {
        render(<TimelineDuration {...defaultProps} />);

        const durationElement = screen.getByText('test.duration.key');
        expect(durationElement).toHaveClass(
            'text-xs',
            'font-medium',
            'transition-colors',
            'duration-300'
        );
    });

    it('should have correct wrapper structure', () => {
        const { container } = render(<TimelineDuration {...defaultProps} />);

        const wrapper = container.firstChild as HTMLElement;
        expect(wrapper).toHaveClass('text-center');
    });

    it('should translate duration key', () => {
        render(
            <TimelineDuration
                {...defaultProps}
                duration="experience.exp1.duration"
            />
        );

        expect(
            screen.getByText('experience.exp1.duration')
        ).toBeInTheDocument();
    });

    it('should match snapshot when selected', () => {
        const { container } = render(
            <TimelineDuration {...defaultProps} isSelected={true} />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot when not selected', () => {
        const { container } = render(
            <TimelineDuration {...defaultProps} isSelected={false} />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
