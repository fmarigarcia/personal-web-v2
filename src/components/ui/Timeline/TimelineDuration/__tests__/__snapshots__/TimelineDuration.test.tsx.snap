// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineDuration should match snapshot when not selected 1`] = `
<div
  class="text-center"
>
  <p
    class="text-xs font-medium transition-colors duration-300 text-stone-600 group-hover:text-stone-950"
  >
    test.duration.key
  </p>
</div>
`;

exports[`TimelineDuration should match snapshot when selected 1`] = `
<div
  class="text-center"
>
  <p
    class="text-xs font-medium transition-colors duration-300 text-stone-950"
  >
    test.duration.key
  </p>
</div>
`;
